flocker/control/testtools.py

```diff
@@ -312,3 +353,22 @@ def deployment_strategy(draw, min_number_of_nodes=1):
         leases={l.dataset_id: l for l in leases},
         persistent_state=persistent_state
     )
+
+
+@composite
+def related_deployments_strategy(draw, number_of_deployments):
+    """
+    A strategy to generate more than 1 deployments that are related.
+
+    Specifically, this ensures that all node uuids are drawn from a common pool
+    for all of the deployments.
+
+    :param int number_of_deployments: The number of deployments to create.
+
+    :returns: A strategy to create ``number_of_deployments`` ``Deployment`` s.
+    """
+    node_uuid_pool = draw(node_uuid_pool_strategy())
+    return tuple(
+        draw(deployment_strategy(node_uuid_pool=node_uuid_pool))
+        for _ in xrange(number_of_deployments)
+    )
```

Comment:
Thanks.

Yea I like this idea in general... if I had more time and thought this code was long-term important I would have done more strategies like this... generate a pool of node uuids, generate a pool of dataset ids, generate a pool of applications, create a bunch of deployments from that, etc.